src/System.Net.Security/src/System/Net/Security/SslConnectionInfo.OSX.cs

```diff
@@ -12,7 +12,38 @@ namespace System.Net.Security
 {
     internal partial class SslConnectionInfo
     {
-        private const ExchangeAlgorithmType EcDheAlgorithm = (ExchangeAlgorithmType)44550;
+        private const int LookupMapCount = 133;
+
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/aa375549(v=vs.85).aspx
+        // CALG_DH_SF
+        private const ExchangeAlgorithmType DiffieHellmanStatic = (ExchangeAlgorithmType)0xAA01;
+        // CALG_ECDH
+        private const ExchangeAlgorithmType EcDhAlgorithm = (ExchangeAlgorithmType)0xAA05;
+        // CALG_ECDH_EPHEM
+        private const ExchangeAlgorithmType EcDheAlgorithm = (ExchangeAlgorithmType)0xAA06;
+
+#if DEBUG
+        static SslConnectionInfo()
+        {
+            Debug.Assert(
+                s_tlsLookup.Count == LookupMapCount,
+                $"Lookup dictionary was of size {s_tlsLookup.Count} instead of {LookupMapCount}");
+
+#if DEBUG_VERBOSE
```

Comment:
If you think it's quick enough, sure.  I was pre-emptively avoiding the "why are you doing Enum.GetValues() on startup?" question :smile:.